File: lib-database-runtime/src/main/java/com/ustadmobile/lib/database/annotation/UmEntity.java
package com.ustadmobile.lib.database.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by mike on 1/21/18.
 */
@Retention(RetentionPolicy.CLASS)
@Target(ElementType.TYPE)
public @interface UmEntity {

    String[] primaryKeys() default {};

    UmIndex[] indices() default {};

    int tableId() default 0;

}

File: lib-database-annotations/src/commonMain/kotlin/com/ustadmobile/lib/annotation/UmEntity.kt
package com.ustadmobile.lib.database.annotation

/**
 * Created by mike on 1/21/18.
 */
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
/*
 Kotlin translation: removed [] as this does not translate to javascript
 */
annotation class UmEntity(/*val primaryKeys: Array<String> = [], val indices: Array<UmIndex> = [], */val tableId: Int = 0)
